scripts/run_inference.py

```diff
@@ -183,13 +183,30 @@ def load(
                 try:
                     if base_model_name_or_path is None:
                         raise ValueError("base_model_name_or_path has to be passed")
-                    base_model = AutoModelForCausalLM.from_pretrained(
-                        base_model_name_or_path,
-                        attn_implementation="flash_attention_2"
-                        if use_flash_attn
-                        else None,
-                        torch_dtype=torch.bfloat16 if use_flash_attn else None,
+
+                    is_quantized = os.path.exists(
+                        os.path.join(base_model_name_or_path, "quantize_config.json")
                     )
+                    if is_quantized:
+                        gptq_config = GPTQConfig(bits=4, exllama_config={"version": 2})
```

Comment:
Can you describe what this is doing? Loading with 4bit GPTQ, what is the `exllama_config` version? It looks like its an exllama kernel?

Reply:
yes this should be loading 4 bit GPTQ with exllama

Reply:
@aluu317 This should also be added to be able to load a base or fine tuned quantized model for inference not just for adapters. This logic should also exist in the else case

Reply:
Done. I added the case for loading a base model and tested with command:

```
python run_inference.py --text "This is a text" --use_flash_attn --model /testing/models/granite-34b-gptq
```

did not throw error.

Not sure about fine tuning a quantized model, maybe we will have that support?

Reply:
agreed im not sure about fine tuning quanitized models...but its nice to have to run inference against base models